Mathlib/Combinatorics/SimpleGraph/Connectivity/WalkCounting.lean

```diff
@@ -169,6 +170,30 @@ instance : Decidable G.Connected := by
 instance instDecidableMemSupp (c : G.ConnectedComponent) (v : V) : Decidable (v ∈ c.supp) :=
   c.recOn (fun w ↦ decidable_of_iff (G.Reachable v w) $ by simp)
     (fun _ _ _ _ ↦ Subsingleton.elim _ _)
+
+lemma exists_odd_component_of_odd_card
+    (ho : Odd (Fintype.card V)) : ∃ (c : ConnectedComponent G), Odd (Nat.card c.supp) := by
```

Comment:
Can you show the stronger statement that the number of odd components is odd?

Reply:
Done:) I did need a little bit more of a specific helper lemma, and I don't think there was a natural place to put it, so I created a new file for that one.